src/sdk/actions.ts

```diff
@@ -130,3 +142,47 @@ async function returnDataToKernel(repoToken: string, stateId: string, output: ob
     },
   });
 }
+
+function createStructuredMetadata(className: string | undefined, logReturn: LogReturn) {
```

Comment:
`LogReturn` looks like a required param here but we treat it as optional below, shouldn't it be made optional here? afaik `LogReturn` is a class and it's not typed as `undefined` in the logger, does the package need updated in this case?

Reply:
Unfortunate copy paste, fixed.